web/src/rendered_markdown.js

```diff
@@ -152,8 +152,14 @@ export const update_elements = ($content) => {
         }
         const message_id = rows.id($message_row);
         const message = message_store.get(message_id);
+        const stream_id = message.stream_id;
 
-        if (message.topic_wildcard_mentioned) {
+        // Highlight the mention only if the user is subscribed to the stream and
+        // is a topic participant.
+        if (
+            message.topic_wildcard_mentioned &&
+            stream_data.is_user_subscribed(stream_id, people.my_current_user_id())
+        ) {
```

Comment:
I think this change is wrong. I confirmed with @alya that `message.topic_wildcard_mentioned` is the correct authoritative source for whether the user was in fact mentioned by this particular `@topic`, and there's no reason to check the user's subscriptions again.

Reply:
In an unsubscribed stream, we don't highlight mentions. So, in an unsubscribed stream, the @topic is treated as you were not mentioned (even though the flag is set)